File: backend/app/schemas/caja.py
from datetime import date, datetime
from pydantic import BaseModel, ConfigDict


class CajaBase(BaseModel):
    numero_economico: str | None = None
    placas: str
    tipo_caja: str | None = None
    marca: str | None = None
    modelo: str | None = None
    anio: int | None = None
    poliza_seguro: str | None = None
    seguro_vigencia: date | None = None
    tarjeta_circulacion: str | None = None
    tarjeta_vigencia: date | None = None
    verificacion: str | None = None
    verificacion_vigencia: date | None = None
    activo: bool = True


class CajaCreate(CajaBase):
    pass


class CajaUpdate(BaseModel):
    numero_economico: str | None = None
    placas: str | None = None
    tipo_caja: str | None = None
    marca: str | None = None
    modelo: str | None = None
    anio: int | None = None
    poliza_seguro: str | None = None
    seguro_vigencia: date | None = None
    tarjeta_circulacion: str | None = None
    tarjeta_vigencia: date | None = None
    verificacion: str | None = None
    verificacion_vigencia: date | None = None
    activo: bool | None = None


class CajaResponse(CajaBase):
    id_caja: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/caja.ts
export type Caja = {
  id_caja: number;
  numero_economico: string | null;
  placas: string;
  tipo_caja: string | null;
  marca: string | null;
  modelo: string | null;
  anio: number | null;
  poliza_seguro: string | null;
  seguro_vigencia: string | null;
  tarjeta_circulacion: string | null;
  tarjeta_vigencia: string | null;
  numero_serie: string | null;
  verificacion: string | null;
  verificacion_vigencia: string | null;
  activo: boolean;
  created_at: string;
  updated_at: string;
};

export type CreateCajaPayload = {
  numero_economico?: string | null;
  placas: string;
  tipo_caja?: string | null;
  marca?: string | null;
  modelo?: string | null;
  anio?: number | null;
  poliza_seguro?: string | null;
  seguro_vigencia?: string | null;
  tarjeta_circulacion?: string | null;
  tarjeta_vigencia?: string | null;
  numero_serie?: string | null;
  verificacion?: string | null;
  verificacion_vigencia?: string | null;
  activo: boolean;
};

export type UpdateCajaPayload = {
  numero_economico?: string | null;
  placas?: string | null;
  tipo_caja?: string | null;
  marca?: string | null;
  modelo?: string | null;
  anio?: number | null;
  poliza_seguro?: string | null;
  seguro_vigencia?: string | null;
  tarjeta_circulacion?: string | null;
  tarjeta_vigencia?: string | null;
  numero_serie?: string | null;
  verificacion?: string | null;
  verificacion_vigencia?: string | null;
  activo?: boolean;
};
